Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.IntegrationTests/FirestoreCache/FirestoreCacheTest.cs

```diff
@@ -0,0 +1,195 @@
+            // an exception.
+            await _fixture.Cache.RefreshAsync(_key);
+
+            // Set it.
+            await _fixture.Cache.SetAsync(_key, _value, 
+                new DistributedCacheEntryOptions()
+                {
+                    SlidingExpiration = TimeSpan.FromSeconds(1200)
+                });
+
+            // Get it.
+            Assert.Equal(_value, await _fixture.Cache.GetAsync(_key));
+
+            // Remove it.
+            await _fixture.Cache.RemoveAsync(_key);
+
+            // Entry does not exist after removing.
+            Assert.Null(await _fixture.Cache.GetAsync(_key));
+        }
+
+        [Fact]
+        void TestSync()
+        {
+            // Entry does not exist before setting.
+            Assert.Null(_fixture.Cache.Get(_key));
+
+            // Makes sure refreshing an entry before it exists doesn't throw
+            // an exception.
+            _fixture.Cache.Refresh(_key);
+
+            // Set it.
+            _fixture.Cache.Set(_key, _value, 
+                new DistributedCacheEntryOptions()
+                {
+                    SlidingExpiration = TimeSpan.FromSeconds(1200)
+                });
+
+            // Get it.
+            Assert.Equal(_value, _fixture.Cache.Get(_key));
+
+            // Remove it.
+            _fixture.Cache.Remove(_key);
+
+            // Entry does not exist after removing.
+            Assert.Null(_fixture.Cache.Get(_key));
+        }
+
+        [Fact]
+        public async Task TestSlidingExpires()
+        {
+            await _fixture.Cache.SetAsync(_key, _value, 
+                new DistributedCacheEntryOptions()
+                {
+                    SlidingExpiration = TimeSpan.FromSeconds(4)
+                });
+            await Task.Delay(3000);
+            // Confirm Refresh() prevents expiration.
+            await _fixture.Cache.RefreshAsync(_key);
+            await Task.Delay(3000);
+            Assert.Equal(_value, await _fixture.Cache.GetAsync(_key));
+            await Task.Delay(3000);
+            // Now, should have expired.
+            Assert.Null(await _fixture.Cache.GetAsync(_key));
+
+        }
+
+        [Fact]
+        public async Task TestAbsoluteExpires()
+        {
+            await _fixture.Cache.SetAsync(_key, _value, 
+                new DistributedCacheEntryOptions()
+                {
+                    AbsoluteExpiration = DateTime.UtcNow.AddSeconds(4),
+                });
+            Assert.Equal(_value, await _fixture.Cache.GetAsync(_key));
+            await Task.Delay(5000);
+            Assert.Null(await _fixture.Cache.GetAsync(_key));
+        }
+
+        [Fact]
+        public async Task TestAbsoluteFromNowExpires()
+        {
+            await _fixture.Cache.SetAsync(_key, _value, 
+                new DistributedCacheEntryOptions()
+                {
+                    AbsoluteExpirationRelativeToNow = TimeSpan.FromSeconds(4)

```

Comment:
Possibly use the "wait 3000, then another 3000" approach here to emphasize the difference between sliding and absolute?

Reply:
Done.